src/battle_util.c

```diff
@@ -2089,7 +2089,7 @@ u8 DoFieldEndTurnEffects(void)
             }
             break;
         case ENDTURN_TRICK_ROOM:
-            if (gFieldStatuses & STATUS_FIELD_TRICK_ROOM && --gFieldTimers.trickRoomTimer == 0)
+            if (gFieldStatuses & STATUS_FIELD_TRICK_ROOM && gFieldTimers.trickRoomTimer > 0 && --gFieldTimers.trickRoomTimer == 0)
```

Comment:
again would vote for 0xFF to be 'infinite' turns otherwise this logic just looks confusing.  Altneratively, maybe a STATUS_FIELD_INFINITE_TRICK_ROOM would be better

Reply:
I'm not sure if I understand the suggestion correctly because right now if you set `B_SET_TRICK_ROOM` it will be infinite. 

I wouldn't mind removing `B_VAR_TERRAIN_TIMER` though. I myself don't see much of a point in it but thought people might like it.

Reply:
Just to clarify the timer doesn't need to be set at all. Setting `gFieldTimers.trickRoomTimer > 0` prevents the field status removal. (something I learned from Skeli during the pledge PR)